packages/api/src/command/medical/tcm-encounter/get-tcm-encounters.ts

```diff
@@ -1,88 +1,109 @@
-    order: [["admitTime", "DESC"]],
-  });
+  const sequelize = TcmEncounterModel.sequelize;
+  if (!sequelize) throw new Error("Sequelize not found");
 
-  const items = rows.map((row: TcmEncounterModel) => {
-    const patient = (row.get("PatientModel") as PatientModel).dataValues.data;
+  const { toItem, fromItem } = pagination;
 
-    /** Hack to get around Sequelize type inference not seeing associations */
-    const encounterData = { ...row.dataValues, PatientModel: undefined };
-    delete encounterData.PatientModel;
+  /**
+   * ⚠️ Always change this query and the count query together.
+   */
+  const queryString = `
+      SELECT tcm_encounter.*, patient.data as patient_data
+      FROM ${tcmEncounterTable} tcm_encounter
+      INNER JOIN ${patientTable} patient 
+      ON tcm_encounter.patient_id = patient.id
+      WHERE tcm_encounter.cx_id = :cxId
+      AND tcm_encounter.admit_time > :afterDate
+      ${toItem ? ` AND tcm_encounter.id >= :toItem` : ""}
+      ${fromItem ? ` AND tcm_encounter.id <= :fromItem` : ""}
+      ORDER BY tcm_encounter.id ${toItem ? "ASC" : "DESC"}
+      LIMIT :count
+    `;
 
-    return {
-      ...encounterData,
-      patientName: patient.firstName + " " + patient.lastName,
-      patientDateOfBirth: patient.dob,
-      patientPhoneNumbers: patient.contact?.map(contact => contact.phone) ?? [],
-      patientStates: patient.address?.map(address => address.state) ?? [],
-    };
-  });
+  /**
+   * Use a type assertion to cast the additional data from the join into the return type.
+   */
+  const rawEncounters = (await sequelize.query(queryString, {
+    model: TcmEncounterModel,
+    mapToModel: true,
+    replacements: {
+      cxId,
+      ...{ afterDate: after ? buildDayjs(after).toDate() : DEFAULT_FILTER_DATE },
+      ...pagination,
+    },
+    type: QueryTypes.SELECT,
+  })) as TcmEncounterQueryData[];
+
+  const encounters = rawEncounters.map(e => ({
+    ...omit(e.dataValues, "patient_data"),
+    patientData: e.dataValues.patient_data,
+  }));
 
-  return items;
+  return sortForPagination(encounters, pagination);
 }
 
 export async function getTcmEncountersCount({
   cxId,
   after,
-}: CountTcmEncountersCmd): Promise<number> {
-  const where: WhereOptions<TcmEncounterModel> = {
-    cxId,
-    admitTime: {
-      [Op.gt]: after ? buildDayjs(after).toDate() : DEFAULT_FILTER_DATE,
-    },
-  };
+}: {
+  cxId: string;
+  after?: string;
+}): Promise<number> {
+  const tcmEncounterTable = TcmEncounterModel.tableName;
 
-  const count = await TcmEncounterModel.count({
-    where,
+  const sequelize = TcmEncounterModel.sequelize;
+  if (!sequelize) throw new Error("Sequelize not found");
+
+  /**
+   * ⚠️ Always change this query and the data query together.
+   */
+  const queryString = `
+    SELECT count(tcm_encounter.id) as count
+    FROM ${tcmEncounterTable} tcm_encounter
+    WHERE tcm_encounter.cx_id = :cxId
+    AND tcm_encounter.admit_time > :afterDate
```

Comment:
ditto

(I'd still have these two lines (where + and) as a `baseFilters` const, but I'll leave that to you, non blocking)